### DISCUSSION TITLE
Issue with fulfillBasicOrder while executing it through a smart contract

### DISCUSSION THREAD

**Discussion by CleanPegasus** [ProjectOpenSea/seaport#639] [Q&A]
I tried to fulfil an order for an NFT with WETH. But when I tried to implement it through a smart contract, I'm getting the following error. I have enough WETH in my account and enough ETH for the gas(Rinkeby).

<img width="504" alt="image" src="https://user-images.githubusercontent.com/42714467/186509680-01caabe6-3b7e-4c50-becc-95f07fd2dcf8.png">

Here is the function I'm trying to execute.

`    function fulfillSeaportOrder() public returns(bool) {

        AdditionalRecipient memory seaportCut = AdditionalRecipient({
            amount: 25000000000000000,
            recipient: payable(0x0000a26b00c1F0DF003000390027140000fAa719)

        });

        AdditionalRecipient memory dead = AdditionalRecipient({
            amount: 1000000000000000,
            recipient: payable(0x000000000000000000000000000000000000dEaD)
        });

        additionalRecipients.push(seaportCut);
        additionalRecipients.push(dead);

        BasicOrderParameters memory order = BasicOrderParameters({
        considerationToken: 0xc778417E063141139Fce010982780140Aa0cD5Ab, 
        considerationIdentifier: 0,
        considerationAmount: 974000000000000000,
        offerer: payable(0x1845aD3465b6BFB4C402bbc682C56bed473D0cC3), // your address
        zone: 0x00000000E88FE2628EbC5DA81d2b3CeaD633E89e,
        offerToken: 0xf5de760f2e916647fd766B4AD9E85ff943cE3A2b, // tokenAddress
        offerIdentifier: 1521045,
        offerAmount: 1, // total amount
        basicOrderType: BasicOrderType.ERC20_TO_ERC721_FULL_OPEN, // ETH_TO_ERC721_FULL_OPEN
        startTime: 1661366912,
        endTime:   1664045312,
        zoneHash: 0x0000000000000000000000000000000000000000000000000000000000000000,
        salt: 317612967258845460796535291342557661168,
        offererConduitKey: 0x0000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f0000,
        fulfillerConduitKey: 0x0000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f0000,
        totalOriginalAdditionalRecipients: 0,
        additionalRecipients: additionalRecipients,
        signature: "0xbe88efc1a4f9b84551918bd688f75728d55926b70d14d0a444d00a580e3a27cd02957e3309508b0cc53ef457ac007d251a678052c67059b7f4feb2994f4c82871c"
        });

        bool status = seaport.fulfillBasicOrder(order);

        return status;
    }`

**Comment by ryanio** [comment]
According to https://sig.eth.samczsun.com/ the custom error `0x8baa579f` you are receiving means `InvalidSignature()`

**Reply by CleanPegasus** [reply]
I'm using the signature of the order from the opensea API. Can you help me out with where I'm going wrong? Thanks in advance.

**Comment by JhinQaQ** [comment]
Hi, I am also trying to call the fulfillBasicOrder function via Etherscan and generate the parameters manually. But I encountered a problem here. Can you tell me how to generate the parameters and the signature?

**Comment by jecombe** [comment]
i've same problem have you solved ?

**Reply by themuli** [reply]
Hey @jecombe did you get this error resolved?

**Reply by jecombe** [reply]
Yep and you ?

**Reply by jecombe** [reply]
if you have a problem please check my last response of @RogerPodacter

**Comment by RogerPodacter** [comment]
Same issue on my end as well!

**Reply by jecombe** [reply]
1: Do not forget ERC721 receiver on your smart contract
2: The function fulFillBasicOrder is payable: `SEAPORT.fulfillBasicOrder{value: msg.value}(order)`
3: Maybe the problem is your signature: you need to get signature with opensea api.
Following the example above, if you write the signature directly in the smart contract, use a hexadecimal value instead of a string.: `signature: hex"be88efc1a4f9b84551918bd688f75728d55926b70d14d0a444d00a580e3a27cd02957e3309508b0cc53ef457ac007d251a678052c67059b7f4feb2994f4c82871c"`.

If I have resolved your bug, please flag it as "Answered". thank's

**Reply by RogerPodacter** [reply]
I still can't figure it out!

Here is my contract:

```solidity
//SPDX-License-Identifier: MIT
pragma solidity >=0.8.0 <0.9.0;

import { Seaport } from "seaport-core/src/Seaport.sol";
import { BasicOrderParameters } from "seaport-types/src/lib/ConsiderationStructs.sol";
import { ERC1155Holder } from "@openzeppelin/contracts/token/ERC1155/utils/ERC1155Holder.sol";

contract SeaportContractBuyTest is ERC1155Holder {
    Seaport public immutable seaport = Seaport(payable(0x00000000000000ADc04C56Bf30aC9d3c0aAF14dC));
    
    function testBuy(BasicOrderParameters calldata orderParams) payable public {
        seaport.fulfillBasicOrder{value: msg.value}(
            orderParams
        );
    }
}
```

I am trying to fulfill the 0.009 offer on [this item](https://opensea.io/assets/ethereum/0xd19d35601c9f4156cc2cfcca42ae4ae4a44acf9a/98817275316469254658835613186317234565061035910961234952676070055051228466648).

The OS API gives me this transaction info:

```json
{
  "function": "fulfillBasicOrder_efficient_6GL6yc((address,uint256,uint256,address,address,address,uint256,uint256,uint8,uint256,uint256,bytes32,uint256,bytes32,bytes32,uint256,(uint256,address)[],bytes))",
  "chain": 1,
  "to": "0x00000000000000adc04c56bf30ac9d3c0aaf14dc",
  "value": 9000000000000000,
  "input_data": {
    "parameters": {
      "considerationToken": "0x0000000000000000000000000000000000000000",
      "considerationIdentifier": "0",
      "considerationAmount": "8325000000000000",
      "offerer": "0xc2172a6315c1d7f6855768f843c420ebb36eda97",
      "zone": "0x004c00500000ad104d7dbd00e3ae0a5c00560c00",
      "offerToken": "0xd19d35601c9f4156cc2cfcca42ae4ae4a44acf9a",
      "offerIdentifier": "98817275316469254658835613186317234565061035910961234952676070055051228466648",
      "offerAmount": "1",
      "basicOrderType": 5,
      "startTime": "1685450679",
      "endTime": "1688129079",
      "zoneHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
      "salt": "24446860302761739304752683030156737591518664810215442929817171426424090205386",
      "offererConduitKey": "0x0000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f0000",
      "fulfillerConduitKey": "0x0000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f0000",
      "totalOriginalAdditionalRecipients": "2",
      "additionalRecipients": [
        {
          "amount": "225000000000000",
          "recipient": "0x0000a26b00c1f0df003000390027140000faa719"
        },
        {
          "amount": "450000000000000",
          "recipient": "0x542430459de4a821c32daa89b00de3f2a8cf43b9"
        }
      ],
      "signature": "0x25ed7878de614c36d1eac15a55b776b7e40e888ff67901478e623489e2c10dca2730d416587b0610c7e5566cb1d98563457af6b95c5d0f9903cb4a9195aeb75b"
    }
  }
}
```

I translate it into the array format I need to pass it as a param in Remix:

```json
[
    "0x0000000000000000000000000000000000000000",
    "0",
    "8325000000000000",
    "0xc2172a6315c1d7f6855768f843c420ebb36eda97",
    "0x004c00500000ad104d7dbd00e3ae0a5c00560c00",
    "0xd19d35601c9f4156cc2cfcca42ae4ae4a44acf9a",
    "98817275316469254658835613186317234565061035910961234952676070055051228466648",
    "1",
    5,
    "1685450679",
    "1688129079",
    "0x0000000000000000000000000000000000000000000000000000000000000000",
    "24446860302761739304752683030156737591518664810215442929817171426424090205386",
    "0x0000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f0000",
    "0x0000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f0000",
    "2",
    [
      [
        "225000000000000",
        "0x0000a26b00c1f0df003000390027140000faa719"
      ],
      [
        "450000000000000",
        "0x542430459de4a821c32daa89b00de3f2a8cf43b9"
      ]
    ],
    "0x25ed7878de614c36d1eac15a55b776b7e40e888ff67901478e623489e2c10dca2730d416587b0610c7e5566cb1d98563457af6b95c5d0f9903cb4a9195aeb75b"
  ]
```

But when I try to buy through the contract in Remix using "Remix VM - Mainnet fork" I get a `NoReentrantCalls` error!

Really appreciate the help here!
